java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/ToolCallBehavior.java

```diff
@@ -135,24 +128,26 @@ public boolean functionEnabled(String pluginName, String functionName) {
         return functionSettings.isEmpty();
     }
 
-    public int maximumAutoInvokeAttempts() {
-        return getSetting("maximumAutoInvokeAttempts", DEFAULT_MAXIMUM_AUTO_INVOKE_ATTEMPTS);
-    }
-
-    protected void setFlag(String key, boolean value) {
-        flags.put(key, value);
-    }
-
-    protected void setSetting(String key, int value) {
-        settings.put(key, value);
+    public int getMaximumAutoInvokeAttempts() {
+        return this.maximumAutoInvokeAttempts;
     }
+    public void configureOptions(
+            ChatCompletionsOptions options,
+            List<FunctionDefinition> functions) {
+        if (functions.isEmpty()) {
+            return;
+        }
 
-    protected boolean getFlag(String key) {
-        return flags.getOrDefault(key, false);
-    }
+        options.setTools(functions.stream()
+//            .filter(function -> {
+//                String[] parts = function.getName().split("_");
+//                String pluginName = parts.length > 0 ? parts[0] : "";
+//                String fnName = parts.length > 1 ? parts[1] : "";
+//                return toolCallBehavior.functionEnabled(pluginName, fnName);
+//            })
+            .map(ChatCompletionsFunctionToolDefinition::new)
+            .collect(Collectors.toList()));
 
-    protected int getSetting(String key, int defaultValue) {
-        return settings.getOrDefault(key, defaultValue);
+        options.setToolChoice(BinaryData.fromString("auto"));
```

Comment:
Yeah. Maybe this configureOptions should be in aiservices.openai...

Reply:
I think ToolCallBehavior should go there, instead.

Reply:
I tend to agree since it ToolCallBehavior is pretty specific to OpenAI. But ToolCallBehavior is used in InvocationContext, so TCB can't be in aiservices.openai without some change to InvocationContext to let it contain arbitrary elements. Which is fine by me. @johnoliver, what do you think?